docs/source/api_reference/data_format.rst

```diff
@@ -0,0 +1,273 @@
+.. _data_format:
+
+File Format Specifications
+==========================
+
+.. toctree::
+    :maxdepth: 1
+    :hidden: false
+
+    Overview
+    Introduction
+    Description
+    Instructions
+    Illustration
+
+Overview
+--------
+This document has the following content:
+
+- Introduction: What is a ``.ts`` file, why and when to use it.
+- Description: What are the individual components of a ``.ts`` file.
+- Instructions: How to create your own ``.ts`` file.
+- Illustrations: A running example to tie up above sections.
+
+Introduction
+------------
+This document formalizes string identifiers used in ``.ts`` file format.
+Encoded in ``utf-8``, ``.ts`` files stores time-series dataset and its corresponding metadata and can be
+opened via any basic editor like notepad for visual inspection.
+String identifiers refer to strings beginning with ``@`` in the file.
+
+``.ts`` files contains information blocks in the following order:
+
+1. All the comments to describe the dataset. (begins with ``#`` in the file)
```

Comment:
I would say "a description block. All lines in this block begin witih ``#``".

In all the blocks, I would also link to a more detailed description of the block, to be given below. E.g., "a description block. All lines in this block begin witih ``#``. For detail description, see section (reference to section"

Reply:
How about:
```
All lines in this block begin with # and provides citations and comments about the dataset. See [:ref:`subsection on description block <comment description>`](https://github.com/alan-turing-institute/sktime/pull/3380#id1) for more details.
```

What all things do we add in the `comment description` subsection? I've mentioned that it's optional and is used for comments and citations. I don't know if we need to write more.

Reply:
the instruction is vague, e.g., "all the comments", this refers to "the" comments, as if the reader should know what comments to put there. Vague descriptions are not good in a tech documment.

Alt suggestion:
```
1. description block
the description contains any number of (contiguous) lines starting with `#`, each `#` followed by an arbitrary (utf-8) sequence of symbols. The `ts` specification does not prescribe any content for the description block, but it is common to include a description of the data set contained, e.g., a full data dictionary.
```